src/BenchmarkDotNet.Core/Exporters/Xml/SimpleXmlWriter.cs

```diff
@@ -0,0 +1,65 @@
+﻿using System;
+using System.IO;
+using System.Xml;
+
+namespace BenchmarkDotNet.Exporters.Xml
+{
+    public class SimpleXmlWriter : IXmlWriter, IDisposable
+    {
+        private XmlWriter writer;
+
+        public SimpleXmlWriter(TextWriter writer, bool indent)
+        {
+            if (writer == null)
+                throw new ArgumentNullException(nameof(writer));
+
+            this.writer = XmlWriter.Create(writer, new XmlWriterSettings() { Indent = indent });
+        }
+
+        public void WriteElementString(string localName, string value)
+        {
+            writer.WriteElementString(localName, value);
+        }
+
+        public void WriteEndDocument()
+        {
+            writer.WriteEndDocument();
+        }
+
+        public void WriteEndElement()
+        {
+            writer.WriteEndElement();
+        }
+
+        public void WriteStartDocument()
+        {
+            writer.WriteStartDocument();
+        }
+
+        public void WriteStartElement(string localName)
+        {
+            writer.WriteStartElement(localName);
+        }
+
+        // Dispose
+        private bool disposedValue = false;
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!disposedValue)
+            {
+                if (disposing)
+                {
+                    writer.Dispose();
+                }
+
+                disposedValue = true;
+            }
+        }
+
+        public void Dispose()
```

Comment:
the implementation of Dispose pattern should simply delegate the responsibility of disposal to the only disposable field in this class (writer). It should take care of things like throwing `ObjectDisposedException` after disposal etc.

So following code will be just enough:
`public void Dispose() => writer.Dispose()`